spec/Data_Parallelism.tex

```diff
@@ -280,9 +284,110 @@ \section{Forall Intents}
 \end{rationale}
 
 
+\section{Task-Private Variables}
+\label{Task_Private_Variables}
+\index{task-private variables}
+\index{shadow variables}
+\index{data parallelism!task-private variables}
+\index{data parallelism!shadow variables}
+\index{statements!forall@\chpl{forall}!task-private variables}
+\index{statements!forall@\chpl{forall}!shadow variables}
+
+A \emph{task-private variable} declared in a forall loop results
+in a separate shadow variable in each task created by the forall
+loop's parallel iterator, as well as a separate shadow variable
+created at the top level of the parallel iterator itself.
+
+The shadow variable is created at the start of the task or the
+iterator and destroyed at the end of the task/iterator.
+Within the lexical scope of the forall statement or expression,
+the variable name references the shadow variable in the task
+that executed the current yield statement, or in the iterator
+if the current yield occurs in the iterator outside any task constructs.
```

Comment:
This sentence was confusing to me at first. What if you had a sentence or two dedicated to the concept of a "top-level" shadow variable, after which you can dedicate a sentence to the "or in the iterator" part of the sentence I'm commenting on?